app/controllers/application_controller.rb

```diff
@@ -5,6 +5,151 @@ class ApplicationController < Sinatra::Base
   configure do
     set :public_folder, 'public'
     set :views, 'app/views'
+    set :sessions, true
+    set :method_override, true
+    set :session_secret, "fwitter_efrain"
   end
 
```

Comment:
This controller is overloaded. Consider separating your concerns by splitting some of these actions into separate controllers (i.e. a users controller and a tweets controller).